pubsub/kafkapubsub/kafka_test.go

```diff
@@ -276,6 +276,120 @@ func TestKafkaKey(t *testing.T) {
 	}
 }
 
+// TestMultiplePartionsWithRebalancing tests use of a topic with multiple
+// partitions, including the rebalancing that happens when a new consumer
+// appears in the group.
+func TestMultiplePartionsWithRebalancing(t *testing.T) {
+	if !setup.HasDockerTestEnvironment() {
+		t.Skip("Skipping Kafka tests since the Kafka server is not available")
+	}
+	const (
+		keyName   = "kafkakey"
+		nMessages = 50
+	)
+	uniqueID := rand.Int()
+	ctx := context.Background()
+
+	// Create a topic with 10 partitions. Using 10 instead of just 2 because
+	// that also tests having multiple claims.
+	topicName := fmt.Sprintf("%s-topic-%d", sanitize(t.Name()), uniqueID)
+	topicCleanup, err := createKafkaTopic(topicName, 10)
+	defer topicCleanup()
+	if err != nil {
+		t.Fatal(err)
+	}
+	topic, err := OpenTopic(localBrokerAddrs, MinimalConfig(), topicName, &TopicOptions{KeyName: keyName})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := topic.Shutdown(ctx); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// Open a subscription.
+	groupID := fmt.Sprintf("%s-sub-%d", sanitize(t.Name()), uniqueID)
+	subOpts := *subscriptionOptions
+	subOpts.KeyName = keyName
+	sub, err := OpenSubscription(localBrokerAddrs, MinimalConfig(), groupID, []string{topicName}, &subOpts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := sub.Shutdown(ctx); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// Send some messages.
+	send := func() {
+		for i := 0; i < nMessages; i++ {
+			m := &pubsub.Message{
+				Metadata: map[string]string{
+					keyName: fmt.Sprintf("key%d", i),
+				},
+				Body: []byte("hello world"),
+			}
+			err = topic.Send(ctx, m)
```

Comment:
`if err := topic.Send(..); err != nil ..`

Or at the very least, don't re-use the `err` from the outer scope.

Reply:
Done.